### PR TITLE
chore: Merge branch 'develop' into v3

### PR DESCRIPTION
**Pull request by tkrugg**
**Summary**

This sync 'develop' with our v3, preparing for the deletion of `v3` and creation of `v4`.

### CODE REVIEW

File: examples/e-commerce/src/tsconfig.app.json [hunk 1]
```diff
@@ -3,7 +3,6 @@
   "compilerOptions": {
     "outDir": "../out-tsc/app",
     "baseUrl": "./",
-    "module": "es2015",
```
Comment (Haroenv) [inline examples/e-commerce/src/tsconfig.app.json]: is this expected?

File: examples/server-side-rendering/src/app/app.component.ts [hunk 2]
```diff
@@ -73,17 +80,16 @@ export class AppComponent {
       ? this.injector.get('request')
       : undefined;
 
-    const searchParameters = parseServerRequest(req);
+    this.searchParameters = parseServerRequest(req);
```
Comment (Haroenv) [inline examples/server-side-rendering/src/app/app.component.ts]: worth testing out on local if this still works as expected. To note that the approach in v4 is different

File: examples/server-side-rendering/src/tsconfig.app.json [hunk 3]
```diff
@@ -3,7 +3,6 @@
   "compilerOptions": {
     "outDir": "../out-tsc/app",
     "baseUrl": "./",
-    "module": "es2015",
```
Comment (Haroenv) [inline examples/server-side-rendering/src/tsconfig.app.json]: same here, removed on purpose?

File: examples/server-side-rendering/src/app/app.component.ts (hunk 2 above)
Comment (tkrugg) [reply examples/server-side-rendering/src/app/app.component.ts]: that's done on my side, but feel free to test again.

File: examples/e-commerce/src/tsconfig.app.json (hunk 1 above)
Comment (tkrugg) [reply examples/e-commerce/src/tsconfig.app.json]: It's the result of the migration to Angular 8.
It was introduced by this PR: https://github.com/algolia/angular-instantsearch/commit/232368b164baa6b3d30c2c6644470c1f11fda133

File: examples/server-side-rendering/src/tsconfig.app.json (hunk 3 above)
Comment (tkrugg) [reply examples/server-side-rendering/src/tsconfig.app.json]: restored

File: examples/e-commerce/src/tsconfig.app.json (hunk 1 above)
Comment (tkrugg) [reply examples/e-commerce/src/tsconfig.app.json]: restored in 2b89b5f

File: examples/server-side-rendering/src/app/app.component.ts (hunk 2 above)
Comment (tkrugg) [reply examples/server-side-rendering/src/app/app.component.ts]: I've moved searchParameters back to the instance here 2b89b5f

**Pull request merged**